compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/QueryDesugar.java

```diff
@@ -1500,7 +1501,8 @@ public void visit(BLangSimpleVarRef bLangSimpleVarRef) {
         // check whether the symbol and resolved symbol are the same.
         // because, lookup using name produce unexpected results if there's variable shadowing.
         if (symbol != null && symbol != resolvedSymbol && !FRAME_PARAMETER_NAME.equals(identifier)) {
-            if (!identifiers.containsKey(identifier)) {
+            if (!identifiers.containsKey(identifier) && (withinLambdaFunc || queryEnv == null
```

Comment:
Shall we add `map.containsKey` as second condition in the `&&` that we we wont access map always

Reply:
Addressed in https://github.com/ballerina-platform/ballerina-lang/commit/b7f888577eff0080b9de31022179665af240b4e6